### PR TITLE
Add layer handle in lattice

### PR DESCRIPTION
**Pull request by chunhtai**
create a layer handle to keep the layer alive

### CODE REVIEW

**Comment by chunhtai** [comment]
Current build is broken, so the this should have existing test coverage if the build pass

File: app_flutter/lib/widgets/lattice.dart [hunk 1]
```diff
@@ -705,12 +708,12 @@ class _RenderLatticeBody extends RenderBox {
     );
   }
 
-  Layer _clipLayer;
+  LayerHandle<Layer> _clipLayerHandle;
```
Comment (dnfield) [inline app_flutter/lib/widgets/lattice.dart:711]:
```suggestion
  final LayerHandle<ClipRectLayer?> _clipLayerHandle = LayerHandle<ClipRectLayer?>();
```

File: app_flutter/lib/widgets/lattice.dart [hunk 2]
```diff
@@ -580,6 +581,8 @@ class _RenderLatticeBody extends RenderBox {
   @override
   void detach() {
     super.detach();
+    _clipLayerHandle.layer = null;
+    _clipLayerHandle = null;
```
Comment (dnfield) [inline app_flutter/lib/widgets/lattice.dart:585]:
```suggestion
    _clipLayerHandle.layer = null;
```

File: app_flutter/lib/widgets/lattice.dart [hunk 3]
```diff
@@ -567,6 +567,7 @@ class _RenderLatticeBody extends RenderBox {
   @override
   void attach(PipelineOwner owner) {
     super.attach(owner);
+    _clipLayerHandle = LayerHandle<Layer>();
```
Comment (dnfield) [inline app_flutter/lib/widgets/lattice.dart:570]:
```suggestion
```

**Review by dnfield — COMMENTED** [review]
Do we not run this as part of customer_testing?

**Comment by chunhtai** [comment]
oops looks like there are more than one breaking change i need to fix, yes. I might as well add this to customer testing

File: app_flutter/lib/widgets/lattice.dart [hunk 4]
```diff
@@ -580,6 +580,7 @@ class _RenderLatticeBody extends RenderBox {
   @override
   void detach() {
     super.detach();
+    _clipLayerHandle.layer = null;
```
Comment (dnfield) [inline app_flutter/lib/widgets/lattice.dart:583]: this should be in dispose, not detach.

**Comment by CaseyHillers** [comment]
Thanks @chunhtai for the PR! I landed this in https://github.com/flutter/cocoon/pull/1274 to unblock Cocoon while you were OOO.
